pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -203,20 +204,22 @@ func (n *NetworkPolicyController) toNamespacedPeerForCRD(peers []v1alpha1.Networ
 	return &controlplane.NetworkPolicyPeer{AddressGroups: getAddressGroupNames(addressGroups)}, addressGroups
 }
 
-// svcRefToPeerForCRD creates an Antrea controlplane NetworkPolicyPeer from
-// ServiceReference in ToServices field. For ANP, we will use the
-// defaultNamespace(policy Namespace) as the Namespace of ServiceReference that
-// doesn't set Namespace.
-func (n *NetworkPolicyController) svcRefToPeerForCRD(svcRefs []v1alpha1.NamespacedName, defaultNamespace string) *controlplane.NetworkPolicyPeer {
+// svcRefToPeerForCRD creates an Antrea controlplane NetworkPolicyPeer from ServiceReferences in ToServices
+// or ToMultiClusterServices field of a crdv1alpha1 NetworkPolicyPeer. For ANP NetworkPolicyPeers, if
+// Namespace is not provided in the ServiceReference, the policy's Namespace will be assumed.
+func (n *NetworkPolicyController) svcRefToPeerForCRD(svcRefs []v1alpha1.NamespacedName, defaultNamespace string, scope v1alpha1.PeerScope) *controlplane.NetworkPolicyPeer {
 	var controlplaneSvcRefs []controlplane.ServiceReference
 	for _, svcRef := range svcRefs {
-		curNS := defaultNamespace
+		svcNS, svcName := defaultNamespace, svcRef.Name
 		if svcRef.Namespace != "" {
-			curNS = svcRef.Namespace
+			svcNS = svcRef.Namespace
+		}
+		if scope == v1alpha1.ScopeClusterSet {
+			svcName = common.ToMCResourceName(svcName)
```

Comment:
Should we extra the `antrea-mc-` first before generate the new name in case that user input `antrea-mc-foo` directly?  If agent actually reuses the logic of `toService` rules, what does this ClusterSet scope means when it's applied to Service like `antrea-mc-*`?

Reply:
No, we should just allow a Service with ServiceImport in toMulticlusterServices.